### ISSUE TITLE
Replace the term 'chain' with 'machine'

### ISSUE THREAD

**Issue by geigerzaehler** [oscoin/radicle#238]
We use the term *chain* throughout the docs and code. The term incorporates a couple of concepts
* a list of radicle expressions
* an associated state
* something that can be updated
* something that can be hosted
* something that can be identifie4d

In my opinion, *chain* does not describe this adequately and is moreover confusing to people from the outside. I would propose using the term *machine* (short for *Radicle state machine*) everywhere we use chain.

cc @clevinson

**Comment by cloudhead** [comment]
:+1:

**Comment by clevinson** [comment]
I'm in favor of replacing "chain" with something that better reflects its uses. Chain is used too often as short-hand for blockchain within the larger p2p / crypto community. However, I see similar issues with "machine", as this usually refers to physical hardware.

Are there other ideas for terms we could use to abbreviate a Radicle RSM? Do people like the name "feed" ?

**Comment by hxrts** [comment]
I've run into this while explaining the product to a few people. I think highlighting the intrinsic connection between the two can actually be quite useful (at least early on) i.e. "machine" is the operational form and "chain" is the product that represents an ordered record of all operations.

**Comment by cloudhead** [comment]
@clevinson "feed" only captures the *input* part of the rsm, not the whole thing.

**Comment by cloudhead** [comment]
How about this:

* **rsm**: the main underlying concept. Includes expressions, states and a head.
* **expression**: an input to the rsm - the primitive in our network.
* **radicle network**: the set of all rsms.
* **state**: materialized view on an rsm.
* **app**: a user-level materialization of one or more **rsm**s.

**Comment by geigerzaehler** [comment]
I agree Radicle state machine should be the full name that disambiguates it from a user’s host machine. But I’d shorten it to “machine” instead of “rsm”. It makes the documentation much easier to read, for example

> Now we can start a radicle REPL that will be able to communicate with the remote machine.

versus

> Now we can start a radicle REPL that will be able to communicate with the remote RSM.

The rest of the terms you proposed, @cloudhead, makes total sense.

I would also like to keep this concrete instead of abstract, though. What I want to address is confusion for outsiders in the documentation. So please take a look at PR #249 and see if the terms you proposed make sense there. I updated it to use the term “app” in the tutorial to build an issues app (instead of an issues chain).

**Comment by cloudhead** [comment]
Hmm I agree it's easier to read but still find it confusing - my brain replaces "machine" with "computer" automatically when it's ambiguous like "remote machine". When it's not ambiguous it's fine, eg. "the machine's state-transition function", or "the machine's inputs".

If we use "radicle state machine/RSM" as the full name and then abbreviate it to "machine" or "state machine" when it makes sense - that should work no? Eg.

> Now we can start a radicle REPL that will be able to communicate with the remote *state* machine.

**Comment by geigerzaehler** [comment]
I think we can actually get rid of the phrase “remote machine”. By default all machines are kind of remote.

> Now we can start a radicle REPL that will be able to communicate with the machine.

or

> Now we can start a radicle REPL that will be able to communicate with the Radicle state machine.

If we need the phrase we can use “remotely hosted radicle state machine (RSM)”. But I did not find an instance in the docs where this is necessary.

**Comment by geigerzaehler** [comment]
I think we have agreement on “chain” not being a good term. Do we also have some agreement on replacing it with “Radicle state machine“, “RSM“ or “machine“ for short? If so I will invest some time in #249 which would be a huge win for our documentation.

**Comment by jameshaydon** [comment]
I agree that "(radicle) state machine" is better. I think "RSM" is confusing because it clashes with "replicated state machine".
